bricks/stm32/main.c

```diff
@@ -189,6 +332,64 @@ mp_obj_t mp_builtin_open(size_t n_args, const mp_obj_t *args, mp_map_t *kwargs)
 }
 MP_DEFINE_CONST_FUN_OBJ_KW(mp_builtin_open_obj, 1, mp_builtin_open);
 
+static mp_obj_t pb_import_loaded_module(size_t n_args, const mp_obj_t *args) {
+
+    // For backwards compatibility, skip imports for legacy single-script data.
+    // TODO: This can be removed once all IDE tools are updated.
+    if (((uint8_t *)mpy_first)[0] == 'M') {
+        return MP_OBJ_NULL;
+    }
+
+    // Try to find the requested module.
+    mpy_info_t *info = mpy_data_find(mp_obj_str_get_str(args[0]));
+    if (!info) {
+        // Not found, so give up.
+        return MP_OBJ_NULL;
+    }
+
+    // Parse the static script data.
+    mp_reader_t reader;
+    pb_reader_user_data_t data;
+    pb_reader_user_data_new(&reader, &data, mpy_data_get_buf(info), info->mpy_size);
+
+    // Create new module and execute in its own context.
+    mp_obj_t module_obj = mp_obj_new_module(mp_obj_str_get_qstr(args[0]));
+    mp_module_context_t *context = MP_OBJ_TO_PTR(module_obj);
+    mp_compiled_module_t compiled_module = mp_raw_code_load(&reader, context);
+    do_execute_raw_code(context, compiled_module.rc, compiled_module.context);
+
+    // Return the module we found.
+    return module_obj;
+}
+
+// Overrides MicroPython's mp_builtin___import__
+mp_obj_t pb_builtin_import(size_t n_args, const mp_obj_t *args) {
+    // Check that it's not a relative import
+    if (n_args >= 5 && MP_OBJ_SMALL_INT_VALUE(args[4]) != 0) {
+        mp_raise_NotImplementedError(MP_ERROR_TEXT("relative import"));
```

Comment:
How hard would it be to implement relative imports?

Reply:
If Pybricks Code only has one folder, what would we use it for? If it is mainly to encourage users to use relative imports, then yes we could just ignore the leading `.` to make it work, but I suppose I'm not sure that's helpful.

If we need more functionality, we could just enable external imports.